Repositories/ColourRepository.cs

```diff
@@ -8,16 +9,10 @@ namespace ColoursTest.Repositories
 {
     public class ColourRepository : IColourRepository
```

Comment:
in all your repositories I would inject the iconfigation and new up the connection as needed rather than in the services, as if you need a repository anywhere else it will also then need to handle creating the connection, whereas inside the repository its all self contained.

also if you ever replace sql with mongo then you would many things and places change things rather than just in the repository itself.

Reply:
How can I have a transaction across multiple tables? My example is the update of a person needs to update the FavouriteColours table and the People table.

Reply:
I would update favourite colours as part of person repository when updating the person.
Think more code first approach to the domain models rather than database first ... what I mean by that is don't design your repositories around the shape of your database but around the shape of your entities or domain models.